tests/queries/0_stateless/03577_temporary_table_as.sql

```diff
@@ -0,0 +1,6 @@
+DROP TABLE IF EXISTS test;
+create table test (a UInt8, b String, c Nullable(Float), date Date) Engine=MergeTree ORDER BY (a,b) PARTITION BY date;
+SET default_temporary_table_engine = 'MergeTree';
+CREATE TEMPORARY TABLE test3 AS test;
```

Comment:
The test looks good, however it might conflict with https://github.com/ClickHouse/clickhouse-private/pull/34982 FYI @tuanpach